tests/components/ledsc/test_config_flow.py

```diff
@@ -0,0 +1,76 @@
+from unittest.mock import patch
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.ledsc.consts import DOMAIN
+
+USER_INPUT = {"host": "127.0.0.1", "port": 8080}
+IMPORT_CONFIG = {"host": "127.0.0.1", "port": 8080}
+RESULT = {
+    "type": "create_entry",
+    "title": f"LedSC server {USER_INPUT['host']}:{USER_INPUT['port']}",
+    "data": USER_INPUT,
+}
+ENTRY_CONFIG = {"host": "127.0.0.1", "port": 8080}
+
+
+async def test_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(
+            "homeassistant.components.ledsc.config_flow.validate_input",
+            return_value=RESULT,
+    ), patch(
+        "homeassistant.components.ledsc.async_setup_entry",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            USER_INPUT,
+        )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == RESULT["title"]
+    assert result2["data"] == USER_INPUT
+
+
+async def test_form_cannot_connect(hass):
+    """Test we handle errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+    )
+
+    with patch(
+            "homeassistant.components.ledsc.config_flow.validate_input",
+            side_effect=ConnectionError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
```

Comment:
```suggestion
    with patch(
            "homeassistant.components.ledsc.config_flow.WebSClient",
            autospe=True,
    ) as mock_websc:
        mock_websc.connect.side_effect = ConnectionError
        result2 = await hass.config_entries.flow.async_configure(
```

Mock out the external library, not code that is part of your integration (please verify first the suggested code works, have not tested it)
Also you should create a fixture out of the library mocking to reduce duplicate code

Reply:
Done. Test works now